crates/text_edit/src/lib.rs

```diff
@@ -92,12 +114,14 @@ impl TextEdit {
 
         let text_size = TextSize::of(&*text);
         let mut total_len = text_size.clone();
+        let mut max_total_len = text_size.clone();
         for indel in &self.indels {
             total_len += TextSize::of(&indel.insert);
             total_len -= indel.delete.len();
+            max_total_len = max(max_total_len, total_len);
         }
 
-        if let Some(additional) = total_len.checked_sub(text_size.into()) {
+        if let Some(additional) = max_total_len.checked_sub(text_size.into()) {
```

Comment:
Instead of having `max_total_len` we can just write this as
```suggestion
        if let Some(additional) = total_len.max(text_size).checked_sub(text_size.into()) {
```
I think

Reply:
My point behind `max_total_len` was to reserve enough space for all intermediary `apply`s that will happen in the loop below, because those intermediary strings could in theory be larger than the last string.

Reply:
Ah ye, I actually made a mistake when looking at this my bad. My proposal isn't equivalent.